nomad/resource_acl_auth_method.go

```diff
@@ -0,0 +1,407 @@
+func resourceACLAuthMethodCreate(d *schema.ResourceData, meta interface{}) error {
+	providerConfig := meta.(ProviderConfig)
+	client := providerConfig.client
+
+	aclAuthMethod, err := generateNomadACLAuthMethod(d)
+	if err != nil {
+		return err
+	}
+
+	// Create our ACL auth method.
+	log.Print("[DEBUG] Creating ACL Auth Method")
+	aclBindingRuleCreateResp, _, err := client.ACLAuthMethods().Create(aclAuthMethod, nil)
+	if err != nil {
+		return fmt.Errorf("error creating ACL Auth Method: %s", err.Error())
+	}
+	log.Printf("[DEBUG] Created ACL Auth Method %q", aclBindingRuleCreateResp.Name)
+	d.SetId(aclBindingRuleCreateResp.Name)
+
+	return resourceACLAuthMethodRead(d, meta)
+}
+
+func resourceACLAuthMethodDelete(d *schema.ResourceData, meta interface{}) error {
+	providerConfig := meta.(ProviderConfig)
+	client := providerConfig.client
+
+	authMethodName := d.Id()
+
+	// Delete the ACL auth method.
+	log.Printf("[DEBUG] Deleting ACL Auth Method %q", authMethodName)
+	_, err := client.ACLAuthMethods().Delete(authMethodName, nil)
+	if err != nil {
+		return fmt.Errorf("error deleting ACL Auth Method %q: %s", authMethodName, err.Error())
+	}
+	log.Printf("[DEBUG] Deleted ACL Auth Method %q", authMethodName)
+
+	d.SetId("")
+
+	return nil
+}
+
+func resourceACLAuthMethodUpdate(d *schema.ResourceData, meta interface{}) error {
+	providerConfig := meta.(ProviderConfig)
+	client := providerConfig.client
+
+	aclBindingRule, err := generateNomadACLAuthMethod(d)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] %#v", aclBindingRule.Config)
+
+	// Perform the in-place update of the ACL auth method.
+	log.Printf("[DEBUG] Updating ACL Auth Method %q", aclBindingRule.Name)
+	_, _, err = client.ACLAuthMethods().Update(aclBindingRule, nil)
+	if err != nil {
+		return fmt.Errorf("error updating ACL Auth Method %q: %s", aclBindingRule.Name, err.Error())
+	}
+	log.Printf("[DEBUG] Updated ACL Auth Method %q", aclBindingRule.Name)
+
+	return resourceACLAuthMethodRead(d, meta)
+}
+
+func resourceACLAuthMethodRead(d *schema.ResourceData, meta interface{}) error {
+	providerConfig := meta.(ProviderConfig)
+	client := providerConfig.client
+
+	authMethodName := d.Id()
+
+	// If the auth method has not been created, the ID will be an empty string
+	// which means we can skip attempting to perform the lookup.
+	if authMethodName == "" {
+		return nil
+	}
+
+	log.Printf("[DEBUG] Reading ACL Auth Method %q", authMethodName)
+	authMethod, _, err := client.ACLAuthMethods().Get(authMethodName, nil)
+	if err != nil {
+		return fmt.Errorf("error reading ACL Auth Method %q: %s", authMethodName, err.Error())
+	}
+	log.Printf("[DEBUG] Read ACL Auth Method %q", authMethod.Name)
+
+	_ = d.Set("name", authMethod.Name)
+	_ = d.Set("type", authMethod.Type)
+	_ = d.Set("token_locality", authMethod.TokenLocality)
+	_ = d.Set("max_token_ttl", authMethod.MaxTokenTTL)
+	_ = d.Set("default", authMethod.Default)
+	_ = d.Set("config", flattenACLAuthMethodConfig(authMethod.Config))
```

Comment:
There's a `StateWriter` helper to handle multiple `Set` like this, but we haven't use it much yet 😅 
https://github.com/hashicorp/terraform-provider-nomad/blob/main/nomad/data_source_scaling_policy.go#L74-L79